test/IIS.FunctionalTests/RequiresIISAttribute.cs

```diff
@@ -8,14 +8,19 @@
 using System.Security.Principal;
 using System.Xml.Linq;
 using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Win32;
 
 namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
 {
     [AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Class | AttributeTargets.Method)]
     public sealed class RequiresIISAttribute : Attribute, ITestCondition
     {
-        private static readonly bool _isMet;
-        public static readonly string _skipReason;
+        private static bool _isMet;
```

Comment:
Neither should this.

Reply:
It has to be to be set in a static constructor.

Reply:
And then you assign it in non-static ctor?

Reply:
I modify them in the non-static ctor. What you recommend assigning in the static constructor then? Something needs to be static.

Reply:
Put feature checks (_poolEnvironmentVariablesAvailable  etc.) into static ctor and skip reason and _isMet calculation into non-static.